filament/include/filament/Material.h

```diff
@@ -69,6 +69,22 @@ class UTILS_PUBLIC Material : public FilamentAPI {
     using ShaderModel = backend::ShaderModel;
     using SubpassType = backend::SubpassType;
 
+    /**
+     * Defines whether a material instance should use UBO batching or not.
+     */
+    enum class UboBatchingMode {
+        /**
+         * For default, it follows the engine settings.
+         * If UBO batching is enabled on the engine and the material domain is SURFACE, it
+         * turns on the UBO batching. Otherwise, it turns off the UBO batching.
+        */
+        DEFAULT,
+        //! Disable the Ubo Batching for this material
+        NO_UBO_BATCHING,
+        //! Enable the Ubo Batching for this material
+        UBO_BATCHING
```

Comment:
This is a public API, but is also interacting with the not-public API (feature flags).

Should UBO batching really be guarded by a feature flag? Or do you imagine that it should be an engine config?